runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -102,7 +102,15 @@ public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
 
             TableDescriptors tableDescriptors = value.getPayload();
             tableDescriptors.getFileDescriptorsMap().forEach((fdName, fileDescriptorProto) -> {
-                fdProtoMap.putIfAbsent(fileDescriptorProto.getName(), fileDescriptorProto);
+                String protoFileName = fileDescriptorProto.getName();
+                // Looks like protobuf file descriptors are randomly truncating the path.
+                // This causes dynamicProtobufSerializer to fail since the full path is needed.
+                if (protoFileName.equals("corfu_options.proto")) {
+                    fdProtoMap.putIfAbsent(protoFileName, fileDescriptorProto);
+                    // Until the truncating issue can be addressed, manually add both paths.
```

Comment:
maybe add a todo to find the real issue for the truncation?

Reply:
people beat me up if I add TODO in my code :(